.github/workflows/make-release.yml

```diff
@@ -0,0 +1,41 @@
+# Copyright 2021-2024 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+name: "Make release"
+
+on:
+  push:
+    tags:
+      - "openssl-*"
+
+jobs:
+  release:
+    runs-on: "releaser"
+    steps:
+    - name: "Checkout"
+      uses: "actions/checkout@v4"
+      with:
+        fetch-depth: 1
+        ref: ${{ github.ref_name }}
+        github-server-url: "https://github.openssl.org/"
+        repository: "openssl/openssl"
+        token: ${{ secrets.GHE_TOKEN }}
+        path: ${{ github.ref_name }}
+    - name: "Prepare assets"
+      run: |
+        cd ${{ github.ref_name }}
+        ./util/mktar.sh
+        mkdir assets && mv ${{ github.ref_name }}.tar.gz assets/ && cd assets
```

Comment:
You might want to figure out the tarball name by other means, such as `echo openssl-*.tar.gz`, at least if this is to be used with 1.1.1 and 1.0.2, where the refname is `OpenSSL_x_y_zl` (x, y, z are numbers, l is a set of letters).

Reply:
Why would we want to? The tarball name matches the ref name, doesn't it?

Reply:
It would not match for old releases. Do we plan to use this workflow on premium releases?

Reply:
Although another option would be to start using openssl-x.y.zl tags for premium releases as well.

Reply:
> Although another option would be to start using openssl-x.y.zl tags for premium releases as well.

"as well" noted.  Yup, we could...

Reply:
> It would not match for old releases. Do we plan to use this workflow on premium releases eventually?

I don't think so. As far as I know we are not going to distribute premium releases via github releases.

Reply:
We do release premium releases as github releases, just on a different github instance.

Reply:
Premium repo does not have releases, we have extended-releases repo with github releases but it has GitHub Actions disabled.

Reply:
I want to say if we want to automate release creating for extended releases this workflow will look differently.

Reply:
Oh, ok.  I'm not sure how it would look different, but fine.
In that case, `{{ github.ref_name }}` is good enough, since that's valid from OpenSSL 3.0 and on.